views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,521 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Previewer Responsive Scale plugin : Scale
+ *
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'ui/hider',
+    'taoTests/runner/plugin',
+    'taoQtiTestPreviewer/previewer/utils/devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/preview-types',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/mobile-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/desktop-devices',
+    'tpl!taoQtiTestPreviewer/previewer/plugins/tools/scale/scale-wrapper',
+    'ui/selecter',
+    'css!taoItemsCss/preview'
+], function (
+    $,
+    _,
+    __,
+    hider,
+    pluginFactory,
+    devices,
+    previewTypesTpl,
+    mobileDevicesTpl,
+    desktopDevicesTpl,
+    scaleWrapperTpl,
+    selecter
+) {
+    'use strict';
+
+
+    return pluginFactory({
+
+        name: 'scale',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var self = this;
+            var orientation = 'landscape';
+            var DEFAULT_TYPE = 'standard';
+            var selectedDevice = 'standard';
+
+            var deviceTypes = [
+                {
+                    value: 'desktop',
+                    label: __('Desktop preview'),
+                    selected: false
+                },
+                {
+                    value: 'mobile',
+                    label: __('Mobile preview'),
+                    selected: false
+                },
+                {
+                    value: 'standard',
+                    label: __('Actual size'),
+                    selected: true
+                },
+            ];
+
+            var screenSize = {
+                width: $(window).innerWidth(),
+                height: $(window).innerHeight()
+            };
+
+            var sizeSettings = {
+                width: 0,
+                height: 0
+            };
+
+            var devicesToControls = {
+                'mobile': '$mobileDevices',
+                'desktop': '$desktopDevices',
+                'standard': '$deviceTypes'
+            };
+
+            var testRunner = this.getTestRunner();
+
+            this.api = {
```

Comment:
the goal of task as to reuse as much as possible from things that were ready. So that part was taken from `taoItems/views/js/preview/preview.js` and those are not exactly side effects,  but encapsulation pattern - way to imitate **private** type methods storing shared data in upper scope.

Reply:
Anyway I moved all that method to separate service https://github.com/oat-sa/extension-tao-testqti-previewer/blob/feature/nex-41/item_preview/views/js/previewer/plugins/tools/scale/responsive-design-testing.js 
and o it would be more convenient and safer to reuse that data.